### DISCUSSION TITLE
7 -> 8 migration - hide keyboard

### DISCUSSION THREAD

**Discussion by ShannyWorkiz** [appium/java-client#1782] [Q&A]
Hey! 
I used to work with java-client version 7.3.0, when moving to 8.0.0 the driver.hideKeyboard() is not accessible anymore.
Does anyone face this issue?

where my drive was set as:
`public static AppiumDriver driver;`

`     public static void initDriver(String className) throws Exception {
        Drivers.setDefaultParams();
        URL url = new URL(props.getProperty("appiumUrl"));
        switch (props.getProperty("platformName").toLowerCase()) {
            case "android":
                driver = Drivers.setAndroidDriver(url, className);
                currentPlatform = "android";
                break;
            case "ios":
                driver = Drivers.setIosDriver(url, className);
                currentPlatform = "ios";
                break;
            default:
                throw new RuntimeException("'platformName' property value is not valid! Got: " + props.getProperty("platformName"));
        }
        logger.info("driver initialized: " + driver);
    }
    //===================================================================================================

    //===================================================================================================

    public static void setDefaultParams(){
        desiredCapabilities.setCapability("platformName", props.getProperty("platformName"));
        desiredCapabilities.setCapability("autoGrantPermissions", true);
        desiredCapabilities.setCapability("sessionTimeout", "5m");
    }
    //===================================================================================================

    //===================================================================================================

    public static AppiumDriver setAndroidDriver(URL url, String className) throws MalformedURLException {
        logger.info("Setting android driver..");
        desiredCapabilities.setCapability("automationName", props.getProperty("androidAutomationName"));
        desiredCapabilities.setCapability("appPackage", props.getProperty("androidAppPackage"));
        desiredCapabilities.setCapability("appActivity", props.getProperty("androidAppActivity"));
        if (props.getProperty("emulator").equalsIgnoreCase("true")) {
            desiredCapabilities.setCapability("deviceName", props.getProperty("mobileName"));
            desiredCapabilities.setCapability("uuid", props.getProperty("uuid"));
            desiredCapabilities.setCapability("avd", props.getProperty("deviceName"));
            desiredCapabilities.setCapability("avdLaunchTimeout", 120000);

        }
        if (props.getProperty("browserstack").equalsIgnoreCase("true")){
            return setAndroidBrowserStackDriver(className);
        }

        String androidAppUrl = new File("src/main/resources/release/app-release.apk").getAbsolutePath().toString();
        logger.info("appUrl is" + androidAppUrl);
        desiredCapabilities.setCapability("app", androidAppUrl);
        return driver = new AndroidDriver(url, desiredCapabilities);
    }
    //===================================================================================================

    //===================================================================================================

    public static AppiumDriver setIosDriver(URL url, String className) throws MalformedURLException{
        logger.info("Setting IOS driver...");
        desiredCapabilities.setCapability("automationName", props.getProperty("iOSAutomationName"));
        logger.info("appUrl is" + props.getProperty("iOSAppLocation"));
        desiredCapabilities.setCapability("app", props.getProperty("iOSAppLocation"));
        desiredCapabilities.setCapability("bundleId", props.getProperty("iOSBundleId"));
        if (props.getProperty("browserstack").equalsIgnoreCase("true")){
            return setAndroidBrowserStackDriver(className);
        }
       return driver = new IOSDriver(url, desiredCapabilities);
    }`

**Comment by valfirst** [comment]
https://github.com/appium/java-client/blob/master/docs/v7-to-v8-migration-guide.md#appiumdriver
You should either use specific driver implementation (e.g. use `IOSDriver` instead of `AppiumDriver`) or cast your driver instance to `io.appium.java_client.HidesKeyboard`
